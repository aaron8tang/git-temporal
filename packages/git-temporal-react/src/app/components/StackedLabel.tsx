import React from 'react';
import { style } from 'app/styles';

export interface StackedLabelProps {
  // This is the text or JSX that gets wrapped in stacked label
  children: string | JSX.Element;
  label: string;
  labelStyle?: object | string;
  onLabelClick?: (evt) => void;
}

const defaultContainerStyle = {
  flexGrow: 0,
  marginRight: 20,
  marginBottom: 10,
};

export const StackedLabel = (props: StackedLabelProps): JSX.Element => {
  return (
    <div style={defaultContainerStyle}>
      <div style={style(props.labelStyle)} onClick={props.onLabelClick}>
        {props.label}
      </div>
      <div>{props.children}</div>
    </div>
  );
};
