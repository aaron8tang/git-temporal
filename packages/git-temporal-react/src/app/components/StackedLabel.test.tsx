import React from 'react';
import { shallow } from 'enzyme';

import { StackedLabel } from './StackedLabel';

describe('components/StackedLabel', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <StackedLabel label="This is the Label">this is the text</StackedLabel>
      );
    });

    test('it should match snapshot (it should have rendered a label and a text element)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
