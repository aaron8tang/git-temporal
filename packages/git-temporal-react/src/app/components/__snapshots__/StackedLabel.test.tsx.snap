// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StackedLabel when rendered with required props and then setting "onLabelClick" prop, it should match snapshot (it should have changed style from first snapshot) 1`] = `
<div
  style={
    Object {
      "flexGrow": 0,
      "marginBottom": 10,
      "marginRight": 20,
    }
  }
>
  <div
    onClick={[MockFunction]}
    style={
      Object {
        "border": "1px solid lightskyblue",
        "borderRadius": 3,
        "cursor": "pointer",
        "userSelect": "none",
      }
    }
  >
    This is the Label
  </div>
  <div>
    this is the text
  </div>
</div>
`;

exports[`components/StackedLabel when rendered with required props and then setting "selected" prop, it should match snapshot (it should have changed style from first snapshot) 1`] = `
<div
  style={
    Object {
      "flexGrow": 0,
      "marginBottom": 10,
      "marginRight": 20,
    }
  }
>
  <div
    style={Object {}}
  >
    This is the Label
  </div>
  <div>
    this is the text
  </div>
</div>
`;

exports[`components/StackedLabel when rendered with required props it should match snapshot (it should have rendered a label and a text element) 1`] = `
<div
  style={
    Object {
      "flexGrow": 0,
      "marginBottom": 10,
      "marginRight": 20,
    }
  }
>
  <div
    style={Object {}}
  >
    This is the Label
  </div>
  <div>
    this is the text
  </div>
</div>
`;
